src/components/likeContainer/LikeContainer.tsx

```diff
@@ -1,14 +1,33 @@
 import { useState } from 'react';
-import type { User } from '../../types';
-import LikedByContainer from '../likedByContainer/LikedByContainer';
 import PostActionBar from '../postActionBar/PostActionBar';
+import LikedByContainer from '../likedByContainer/LikedByContainer';
+import { MOCK_DATA, SAMPLES_PROFILE_PICTURE } from '../../mockText';
+import { classes } from './LikeContainer.st.css';
+import type { User } from '../../types';
 
 const LikeContainer = () => {
-    const [usersWhoLiked, setUsersWhoLiked] = useState<User[]>([]);
-    const likePost = () => {}; // unshift you to the state
+    const [usersWhoLiked, setUsersWhoLiked] = useState<User[]>([
+        { userName: MOCK_DATA.SAMPLE_NAME, profilePicture: SAMPLES_PROFILE_PICTURE[2] },
+    ]);
+
+    const likePost = (user: User) => {
+        let currentUsersWhoLiked = [...usersWhoLiked];
+        let indexOfUser: number = -1;
+        for (let i = 0; i < usersWhoLiked.length; i++) {
+            if (usersWhoLiked[i].userName === user.userName) {
+                indexOfUser = i;
+            }
+        }
```

Comment:
``usersWhoLiked.filter``

Reply:
I need to know first if the user is in the array.
 if true => remove it by index
else => add it